### PR TITLE
does not cancel endpoint reconciliation on undrained pods on deletion events

### PR DESCRIPTION
**Pull request by xh3b4sd**
Towards https://github.com/giantswarm/giantswarm/issues/3782.

### CODE REVIEW

File: service/controller/v14/resource/endpoint/current.go [hunk 1]
```diff
@@ -20,33 +17,20 @@ func (r *Resource) GetCurrentState(ctx context.Context, obj interface{}) (interf
 
 	var serviceName string
 	{
-		r.logger.LogCtx(ctx, "level", "debug", "message", "looking for annotations on pod")
+		r.logger.LogCtx(ctx, "level", "debug", "message", "finding annotations")
```
Comment (xh3b4sd) [inline service/controller/v14/resource/endpoint/current.go]: Improving logs here.

File: service/controller/v14/resource/endpoint/current.go [hunk 2]
```diff
@@ -20,33 +17,20 @@ func (r *Resource) GetCurrentState(ctx context.Context, obj interface{}) (interf
...
-			return nil, microerror.Mask(err)
-		}
-		if !isDrained {
-			r.logger.LogCtx(ctx, "level", "debug", "message", "cannot finish deletion of pod due to undrained status")
-			resourcecanceledcontext.SetCanceled(ctx)
-			finalizerskeptcontext.SetKept(ctx)
-			r.logger.LogCtx(ctx, "level", "debug", "message", "canceling resource")
-
-			return nil, nil
-		}
```
Comment (xh3b4sd) [inline service/controller/v14/resource/endpoint/current.go]: Removing this fixes https://github.com/giantswarm/giantswarm/issues/3782. I cannot recall why we added it, but it prevents terminating pods from getting their broken IPs being removed from endpoints. This works now. The API accessibility issues we perceived as described in the postmortems are all gone now. I am very pleased with this.

**Review by MarcelMue — APPROVED** [review]
Fine with me

File: service/controller/v14/resource/endpoint/current.go (hunk 2 above)
Comment (kopiczko) [reply service/controller/v14/resource/endpoint/current.go]: I don't follow there. This is something based on annotations I don't really remember. It's also used in endpoint resource.

Comment (xh3b4sd) [reply service/controller/v14/resource/endpoint/current.go]: This here is the endpoint resource.

Comment (kopiczko) [reply service/controller/v14/resource/endpoint/current.go]: LoL :D 
OK, so these functions should be also removed from key package.

Comment (xh3b4sd) [reply service/controller/v14/resource/endpoint/current.go]: Good catch. Removed `IsPodDeleted` from key package. `IsPodDraind` is still used in the `pod` resource which we use for draining.

**Pull request merged**